### PR TITLE
Kokkos: updating package list, maintainers and minimum cmake version

### PR DESCRIPTION
**Pull request by lucbv**
Performing some maintenance in the package

1. adding info for release version 3.5.00
2. updating the minimum cmake required version to 3.16 or higher (matches CMakeLists.txt)
3. swapping David for Christian as maintainer

### CODE REVIEW

**Comment by lucbv** [comment]
@crtrott @dalg24 and @janciesko
You guys might want to check this PR before it merges, also you may want to set yourselves as reviewers?
Also I am seeing that the GitHub handle of Jan is different from the maintainer one in the package...
This is blocking PR #28409

**Comment by janciesko** [comment]
LGTM. Yes, @lucbv, please update to the GitHub handle to "janciesko". Thanks for updating the package Luc.

File: var/spack/repos/builtin/packages/kokkos/package.py
Comment (dalg24) [inline var/spack/repos/builtin/packages/kokkos/package.py:22]: You might want to pick someone who actually uses Spack.

**Review by dalg24 — APPROVED** [review]
Looks good.
I didn't verify the sha256.

File: var/spack/repos/builtin/packages/kokkos/package.py
Comment (lucbv) [reply var/spack/repos/builtin/packages/kokkos/package.py:22]: Honestly I put Christian to motivate him to find someone at the next developers meeting, I don't feel it's my decision to make...

**Comment by lucbv** [comment]
@tgamblin do you know if the CI is verifying the checksum as part of the testing procedure?

**Review by crtrott — APPROVED** [review]

**Comment by lucbv** [comment]
@spack/merge-to-develop can we merge this?

**Comment by tgamblin** [comment]
@lucbv: it isn't, but that's a good idea that I'd like to get in there.  Ideally we could validate it by fetching from `https` in CI.

**Review by tgamblin — APPROVED** [review]

**Pull request merged**
